src/aggregator/aggregator/generic_elem.go

```diff
@@ -753,12 +759,10 @@ func (e *GenericElem) processValue(
 				// We currently only support first-order derivative transformations so we only
 				// need to keep one value. In the future if we need to support higher-order
 				// derivative transformations, we need to store an array of values here.
-				if !math.IsNaN(curr.Value) {
-					if flushState.consumedValues == nil {
-						flushState.consumedValues = make([]float64, len(e.aggTypes))
-					}
-					flushState.consumedValues[aggTypeIdx] = curr.Value
+				if flushState.consumedValues == nil {
```

Comment:
this guard was dumb. it just causes a potential nil pointer when accessing the previous `consumedValues` above. instead of having to add a branch for nil, just add NaN which the code can already handle anyways.